apps/saleor-app-checkout/backend/payments/providers/adyen/utils.ts

```diff
@@ -44,12 +55,209 @@ export const getLineItems = (lines: OrderFragment["lines"]): Types.checkout.Line
 export const createEventUniqueKey = (event: TransactionEventFragment) =>
   [event.name, event.reference].join();
 
-export const getAmountAfterRefund = (money: MoneyFragment, refundAmount: number) => {
-  const chargedAmount = money.amount - refundAmount;
+export const encodeBasicAuth = (username: string, password: string) => {
+  return Buffer.from(username + ":" + password, "ascii").toString("base64");
+};
+
+export const verifyBasicAuth = (username: string, password: string, token: string) => {
+  const credentials = encodeBasicAuth(username, password);
+  return token === `Basic ${credentials}`;
+};
+
+export const getNotificationStatus = (
+  notification: Types.notification.NotificationRequestItem
+): TransactionStatus => {
+  const { eventCode, success } = notification;
+  if (
+    failedEvents.has(eventCode) ||
+    success === Types.notification.NotificationRequestItem.SuccessEnum.False
+  ) {
+    return "FAILURE";
+  }
+
+  return "SUCCESS";
+};
+
+export type TransactionAmounts = Pick<
+  TransactionUpdateInput,
+  "amountRefunded" | "amountAuthorized" | "amountCharged" | "amountVoided"
+>;
+
+const nonNegative = (amount: number) => (amount < 0 ? 0 : amount);
```

Comment:
Probably doesn't matter but this function fails to do its job for `-0` 🙃  Surprisingly, the below change fixes the problem:
```suggestion
const nonNegative = (amount: number) => (amount <= 0 ? 0 : amount);
```

Reply:
ah yes, the quirky `-0` 😅  - thanks good catch